apps/asap-server/test/fixtures/users.fixtures.ts

```diff
@@ -6,97 +6,91 @@ import {
   FetchUsersQuery,
 } from '../../src/autogenerated-gql/graphql';
 
-export const getGraphqlResponseFetchUsers = (): {
-  data: FetchUsersQuery;
-} => ({
-  data: {
-    queryUsersContentsWithTotal: {
-      total: 2,
-      items: [
-        getGraphQLUser(),
-        {
-          id: 'user-id-2',
-          created: '2020-09-23T20:45:22Z',
-          lastModified: '2020-10-26T15:33:18Z',
-          version: 42,
-          flatData: {
-            biography: 'some biography',
-            institution: 'some institution',
-            jobTitle: 'some job title',
-            onboarded: true,
-            orcidLastModifiedDate: null,
-            orcidLastSyncDate: null,
-            orcidWorks: [
-              {
-                doi: 'test-doi',
-                id: '987-654-321',
-                lastModifiedDate: '2020-10-26T15:33:18Z',
-                publicationDate: {},
-                type: 'BOOK',
-                title: 'orcid work title 2',
-              },
-            ],
-            reachOut: 'some reach out',
-            responsibilities: 'some responsibilities',
-            researchInterests: 'some research interests',
-            city: 'some city',
-            country: 'some country',
-            contactEmail: 'some@contact.email',
-            degree: 'some degree',
-            orcid: 'orcid',
-            avatar: null,
-            email: 'iwillbeback@arnold.com',
-            firstName: 'Arnold',
-            lastModifiedDate: null,
-            lastName: 'Schwatzneger',
-            questions: [],
-            expertiseAndResourceTags: [],
-            expertiseAndResourceDescription: 'Amazing person',
-            social: [
-              {
-                github: 'awesome',
-                googleScholar: null,
-                linkedIn: null,
-                researcherId: null,
-                researchGate: null,
-                twitter: null,
-                website1: null,
-                website2: null,
-              },
-            ],
-            teams: [
+export const getSquidexUsersGraphqlResponse = (
```

Comment:
do you need you generators (this one and the ones below) to really accept the input args? Not that it is invalid, i just think it would be nice to keep the fixtures clean, use nested ones but never have to alter them too much. 

What I would normally recommend is that any changes to fixtures for the tests should be done after the data is generated, inside the test, through the update of properties, like this: https://github.com/yldio/asap-hub/blob/93ac40b96a40f4a0ea366b17b1800f1661016e31/apps/asap-server/test/controllers/calendars.test.ts#L50-L51

I find it much easier to read, but it is possible that I am not seeing the exact reason why you have this here

Reply:
I prefer not to mutate objects/data. I prefer seeing getSquidexUsersGraphqlResponse({ total: 0 }) or getGraphQLUser({ id: 'some-new-id'})


Let's have a call to discuss?